File: app/src/main/java/com/example/breezil/favnews/ui/SettingsActivity.java
package com.example.breezil.favnews.ui;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.support.design.widget.BottomNavigationView;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.Menu;
import android.view.MenuItem;

import com.example.breezil.favnews.R;
import com.example.breezil.favnews.databinding.ActivitySettingsBinding;
import com.example.breezil.favnews.utils.helpers.BottomNavigationHelper;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.DaggerAppCompatActivity;
import dagger.android.support.HasSupportFragmentInjector;

public class SettingsActivity extends DaggerAppCompatActivity implements HasSupportFragmentInjector {

    ActivitySettingsBinding binding;

    @Inject
    DispatchingAndroidInjector<Fragment> dispatchingAndroidInjector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_settings);

        setupBottomNavigation();

    }



    private void setupBottomNavigation() {

        BottomNavigationHelper.disableShiftMode(binding.bottomNavViewBar);


        Menu menu = binding.bottomNavViewBar.getMenu();
        MenuItem menuItem= menu.getItem(2);
        menuItem.setChecked(true);


        /*
         * here sets the navigations to its corresponding activities
         */
        binding.bottomNavViewBar.setOnNavigationItemSelectedListener(item -> {
            switch (item.getItemId()){

                case R.id.main:
                    startActivity(new Intent(SettingsActivity.this,MainActivity.class));
                    break;
                case R.id.search:
                    startActivity(new Intent(SettingsActivity.this,SearchActivity.class));
                    break;
                case R.id.settings:

                    break;
            }


            return false;
        });

    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/java/com/example/breezil/fave/ui/preference/SettingsActivity.kt
package com.example.breezil.fave.ui.preference

import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment

import com.example.breezil.fave.R
import com.example.breezil.fave.databinding.ActivitySettingsBinding
import com.example.breezil.fave.ui.BaseActivity
import com.example.breezil.fave.ui.explore.SearchActivity
import com.example.breezil.fave.ui.main.MainActivity
import com.example.breezil.fave.utils.helpers.BottomNavigationHelper

import javax.inject.Inject

import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector

class SettingsActivity : BaseActivity(), HasSupportFragmentInjector {

    lateinit var binding: ActivitySettingsBinding

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings)

        setupBottomNavigation()
        supportActionBar!!.title = "Preference"

        binding.aboutText.setOnClickListener { startActivity(Intent(this, AboutActivity::class.java)) }

    }


    private fun setupBottomNavigation() {

        BottomNavigationHelper.disableShiftMode(binding.bottomNavViewBar)


        val menu = binding.bottomNavViewBar.menu
        val menuItem = menu.getItem(2)
        menuItem.isChecked = true


        /*
         * here sets the navigations to its corresponding activities
         */
        binding.bottomNavViewBar.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {

                R.id.main -> {
                    startActivity(Intent(this@SettingsActivity, MainActivity::class.java))
                    finish()
                }
                R.id.search -> {
                    startActivity(Intent(this@SettingsActivity, SearchActivity::class.java))
                    finish()
                }
                R.id.settings -> {
                }
            }


            false
        }

    }

    override fun supportFragmentInjector(): AndroidInjector<Fragment>? {
        return dispatchingAndroidInjector
    }
}
